schematic_api/api/routes.py

```diff
@@ -258,43 +259,16 @@ def get_manifest_route(schema_url: str, use_annotations: bool, dataset_ids=None,
     # call config_handler()
     config_handler(asset_view = asset_view)
     
-    temp_path = get_temp_model_path(schema_url=schema_url)
-
-    # Gather all data_types to make manifests for.
-    all_args = connexion.request.args
-    args_dict = dict(all_args.lists())
-    data_type = args_dict['data_type']
-
-    # Gather all dataset_ids
-    try:
-        dataset_ids = args_dict['dataset_id']
-    except:
-        pass
-    
-    if dataset_ids:
-        # Check that the number of submitted data_types matches
-        # the number of dataset_ids (if applicable)
-        len_data_types = len(data_type)
-        len_dataset_ids = len(dataset_ids)
-        
-        try:
-            len_data_types == len_dataset_ids
-        except:
-            raise ValueError(
-                    f"There is a mismatch in the number of data_types and dataset_id's that "
-                    f"submitted. Please check your submission and try again."
-                )
-        
-        # Raise an error if used in conjunction with datatype = 'all_manifests'
-        try:
-            data_type[0] != 'all manifests'
-        except:
-            raise ValueError(
-                    f"When submitting 'all manifests' as the data_type cannot also submit dataset_id. "
-                    f"Please check your submission and try again."
-                )
-      
-    all_results = ManifestGenerator.create_manifests(path_to_data_model=schema_url, output_format=output_format, data_types=data_type, title=title, access_token=access_token, dataset_ids=dataset_ids, strict=strict_validation, use_annotations=use_annotations)
+    all_results = ManifestGenerator.create_manifests(path_to_data_model=schema_url, output_format=output_format, data_types=data_type, title=title, access_token=access_token, dataset_ids=dataset_id, strict=strict_validation, use_annotations=use_annotations)
+   
+    # return an excel file if output_format is set to "excel"
+    if output_format == "excel":
+        # should only contain one excel spreadsheet path
+        result = all_results[0]
+        dir_name = os.path.dirname(result)
+        file_name = os.path.basename(result)
+        mimetype='application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'
+        return send_from_directory(directory=dir_name, path=file_name, as_attachment=True, mimetype=mimetype, max_age=0)
```

Comment:
I'm a little confused here. It looks like you are still using create_mainfests implying you cna get more than one manifest from this endpoint, but the arugment names are dataset_id and data_type. Shouldn't those be plural?

Reply:
Yeah, they should be plural ideally. But if we want to change them to plural, we would also need to change the query parameters defined in `api.yml`. If we change that, we would also need to inform DFA, DCA so that they could change how they are calling schematic. Let me open a new issue for it and have them in a different PR. For this issue, I think it would be best if we just keep it simply for moving relevant parts from api to schematic library. 

See issue: https://sagebionetworks.jira.com/browse/FDS-1666